diff --git a/commands/metamask.js b/commands/metamask.js
index b32c62856e518052c2b9e837ceb5c247084152f3..b0d774026d0a89dcbb425b25e8eb4dc0ed9cc4fe 100644
--- a/commands/metamask.js
+++ b/commands/metamask.js
@@ -1,5 +1,6 @@
 const log = require('debug')('synpress:metamask');
 const playwright = require('./playwright');
+const sleep = require('util').promisify(setTimeout);
 
 const {
   onboardingWelcomePageElements,
@@ -358,6 +359,18 @@ const metamask = {
     await module.exports.closePopupAndTooltips();
     await playwright.waitAndClick(mainPageElements.accountMenu.button);
     if (typeof accountNameOrAccountNumber === 'number') {
+      const page = await playwright.metamaskWindow();
+      await playwright.waitFor(mainPageElements.accountMenu.accountButton(1), page);
+      for (let i = 2; i <= accountNameOrAccountNumber; i++) {
+        if (!(await page.locator(mainPageElements.accountMenu.accountButton(i)).count())) {
+          await playwright.waitAndClick(
+            mainPageElements.accountMenu.createAccountButton,
+          );
+          await playwright.waitAndClick(mainPageElements.createAccount.createButton);
+          await playwright.waitAndClick(mainPageElements.accountMenu.button);
+        }
+      }
+
       await playwright.waitAndClick(
         mainPageElements.accountMenu.accountButton(accountNameOrAccountNumber),
       );
@@ -817,6 +830,7 @@ const metamask = {
     return true;
   },
   async confirmTransaction(gasConfig) {
+    
     let txData = {};
     const notificationPage = await playwright.switchToMetamaskNotification();
     if (gasConfig) {
diff --git a/synpress.js b/synpress.js
index cf7b4ecc9eee4232fe4e94902b161ddb4f015656..a375c1074f0e19e1daea84004254c7ed0cca9df5 100755
--- a/synpress.js
+++ b/synpress.js
@@ -99,7 +99,10 @@ program
     'specify mocha reporter options, separate multiple values with comma',
   )
   // dashboard
-  .option('-cid, --ciBuildId', '[dashboard] add custom ci build id to the run')
+  .option(
+    '--ci-build-id <id>',
+    '[dashboard] the unique identifier for a run on your CI provider. typically a "BUILD_ID" env var. this value is automatically detected for most CI providers'
+  )
   .option(
     '-r, --record',
     '[dashboard] record video of tests running after setting up your project to record',