diff --git a/dist/index.edge.js b/dist/index.edge.js
index 65621c51eb74e6e8768dd513cdba31eb44997861..06f30167966f3e947626159d82bfdd5050baeaa5 100644
--- a/dist/index.edge.js
+++ b/dist/index.edge.js
@@ -18621,8 +18621,8 @@ function isCustomFontsOptions(value) {
 }
 
 // src/index.edge.ts
-import resvg_wasm from "./resvg.wasm?module";
-import yoga_wasm from "./yoga.wasm?module";
+import resvg_wasm from "./resvg.wasm";
+import yoga_wasm from "./yoga.wasm";
 
 // src/emoji/index.ts
 var U200D = String.fromCharCode(8205);
@@ -18846,22 +18846,13 @@ async function render(satori, resvg, opts, defaultFonts, element) {
 // src/index.edge.ts
 var initializedResvg = initWasm(resvg_wasm);
 var initializedYoga = initYoga(yoga_wasm).then((yoga2) => Rl(yoga2));
-var fallbackFont = fetch(new URL("./noto-sans-v27-latin-regular.ttf", import.meta.url)).then((res) => res.arrayBuffer());
 var ImageResponse = class extends Response {
   constructor(element, options = {}) {
     const result = new ReadableStream({
       async start(controller) {
         await initializedYoga;
         await initializedResvg;
-        const fontData = await fallbackFont;
-        const fonts = [
-          {
-            name: "sans serif",
-            data: fontData,
-            weight: 700,
-            style: "normal"
-          }
-        ];
+        const fonts = [];
         const result2 = await render(wl, resvg_wasm_exports, options, fonts, element);
         controller.enqueue(result2);
         controller.close();