diff --git a/node_modules/next-dev-https/src/server.js b/node_modules/next-dev-https/src/server.js
index c10643e..b7f25fb 100644
--- a/node_modules/next-dev-https/src/server.js
+++ b/node_modules/next-dev-https/src/server.js
@@ -256,13 +256,6 @@ function startServer(opts) {
         reject(err);
       }
     });
-    /** @type { ReturnType<import("next/dist/server/next").NextServer["getUpgradeHandler"]> } */
-    let upgradeHandler;
-    if (!opts.dev) {
-      server.on("upgrade", (req, socket, upgrade) => {
-        upgradeHandler(req, socket, upgrade);
-      });
-    }
     server.on("listening", () => {
       const addr = server.address();
       const hostname =
@@ -277,7 +270,6 @@ function startServer(opts) {
         port: addr && typeof addr === "object" ? addr.port : port,
       });
       requestHandler = app.getRequestHandler();
-      upgradeHandler = app.getUpgradeHandler();
       resolve(app);
     });
     server.listen(port, opts.hostname);
