diff --git a/support/index.js b/support/index.js
index 739591edc103607bef57996054a90a5d0015b41c..926c82fb1782c416a76d6a151e13cfc0d7264d74 100644
--- a/support/index.js
+++ b/support/index.js
@@ -1,4 +1,4 @@
-import './commands';
+import './commands'
 Cypress.on('window:before:load', win => {
   cy.stub(win.console, 'error').callsFake(message => {
     cy.now('task', 'error', message);
@@ -8,9 +8,3 @@ Cypress.on('window:before:load', win => {
     cy.now('task', 'warn', message);
   });
 });
-
-before(() => {
-  cy.setupMetamask()
-});
-
-
