diff --git a/dist/chunk-S7EOD33A.js b/dist/chunk-S7EOD33A.js
index 9dc793378a2d5ff3233f131a4e68a6f9a677e822..baa66bca403ae548045623a1f24157e9b050f548 100644
--- a/dist/chunk-S7EOD33A.js
+++ b/dist/chunk-S7EOD33A.js
@@ -3694,11 +3694,13 @@ export {
   useAddress,
   useChainId,
   useTransactionStore,
+  useRecentTransactions,
   cssObjectFromTheme,
   cssStringFromTheme,
   RainbowKitProvider,
   dialogContent,
   dialogContentMobile,
+  useClearRecentTransactions,
   DesktopOptions,
   MobileOptions,
   ConnectButton
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 3ac88fac1f5d7907f5c9896f56986d91d5971c4b..847618034559d368d9c8acff69c5dbb1fb7718b8 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -4,6 +4,9 @@ export { getDefaultWallets } from './wallets/getDefaultWallets';
 export { getWalletConnectConnector } from './wallets/getWalletConnectConnector';
 export { connectorsForWallets } from './wallets/connectorsForWallets';
 export { useAddRecentTransaction } from './transactions/useAddRecentTransaction';
+export { useClearRecentTransactions } from './transactions/useClearRecentTransactions';
+export { useRecentTransactions } from './transactions/useRecentTransactions';
+export { useTransactionStore } from './transactions/TransactionStoreContext';
 export type { Wallet, WalletList } from './wallets/Wallet';
 export type { Chain } from './components/RainbowKitProvider/RainbowKitChainContext';
 export type { Theme } from './components/RainbowKitProvider/RainbowKitProvider';
diff --git a/dist/index.js b/dist/index.js
index 70e35f87d052b107a7b48d7a410139cd1eac6402..b11edccd77eb5df2ceb3966feb8cd6e05c853638 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -18,6 +18,8 @@ import {
   isMobile,
   useAddress,
   useChainId,
+  useClearRecentTransactions,
+  useRecentTransactions,
   useTransactionStore
 } from "./chunk-S7EOD33A.js";
 import {
@@ -611,5 +613,8 @@ export {
   lightTheme,
   midnightTheme,
   useAddRecentTransaction,
+  useClearRecentTransactions,
+  useRecentTransactions,
+  useTransactionStore,
   wallet
 };
